eth_account/_utils/signing.py

```diff
@@ -40,14 +41,25 @@ def sign_transaction_dict(eth_key, transaction_dict):
 
 
 # watch here for updates to signature format: https://github.com/ethereum/EIPs/issues/191
-def signature_wrapper(message, version=b'E'):
-    assert isinstance(message, bytes)
-    if version == b'E':
+@curry
+def signature_wrapper(message, signature_version, version_specific_data):
+    if not isinstance(message, bytes):
+        raise TypeError("Message is not of the type {}, expected bytes".format(type(message)))
+
+    if signature_version == b'E':
         preamble = b'\x19Ethereum Signed Message:\n'
         size = str(len(message)).encode('utf-8')
         return preamble + size + message
+    elif signature_version == b'\x00':
+        wallet_address = to_bytes(hexstr=version_specific_data)
+        if len(wallet_address) != 20:
+            raise TypeError("Invalid Wallet Address: {}".format(version_specific_data))
+        wrapped_message = b'\x19' + signature_version + wallet_address + message
+        return wrapped_message
     else:
-        raise NotImplementedError("Only the 'Ethereum Signed Message' preamble is supported")
+        raise NotImplementedError(
+            "Only EIP 191 type signature wrapping is currently supported"
```

Comment:
This message should probably include the hex representation of the `signature_version` and potentially an explicit list of the hex representations of the versions that are supported.